File: sdks/php/src/PlanOverrides/Requests/ListPlanOverridesRequest.php
<?php

namespace NovaBilling\PlanOverrides\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class ListPlanOverridesRequest extends JsonSerializableType
{
    /**
     * @var ?string $customerId
     */
    public ?string $customerId;

    /**
     * @var ?string $planId
     */
    public ?string $planId;

    /**
     * @var ?float $page
     */
    public ?float $page;

    /**
     * @var ?float $limit
     */
    public ?float $limit;

    /**
     * @param array{
     *   customerId?: ?string,
     *   planId?: ?string,
     *   page?: ?float,
     *   limit?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->customerId = $values['customerId'] ?? null;
        $this->planId = $values['planId'] ?? null;
        $this->page = $values['page'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/PlanOverrides/Requests/ListPlanOverridesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListPlanOverridesRequest
{
    [JsonIgnore]
    public string? CustomerId { get; set; }

    [JsonIgnore]
    public string? PlanId { get; set; }

    [JsonIgnore]
    public double? Page { get; set; }

    [JsonIgnore]
    public double? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
